delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/model/IdSegment.scala

```diff
@@ -0,0 +1,88 @@
+package ch.epfl.bluebrain.nexus.delta.sdk.model
+
+import ch.epfl.bluebrain.nexus.delta.rdf.IriOrBNode.Iri
+import ch.epfl.bluebrain.nexus.delta.rdf.jsonld.context.{ContextValue, JsonLdContext}
+import ch.epfl.bluebrain.nexus.delta.sdk.model.IdSegment.{IriSegment, ResourceRefSegment, StringSegment}
+import ch.epfl.bluebrain.nexus.delta.sdk.model.projects.{ApiMappings, Project}
+
+/**
+  * A segment from the positional API that should be an Id
+  */
+sealed trait IdSegment extends Product with Serializable { self =>
```

Comment:
As segment comes from akka-http, something like RawId would maybe be better ?

```case class RawId[A](value: A)```

And then we would be two methods in the companion object:

```
def parseIri[A](id: Raw[A])(implicit iriDecoder: IriDecoder[A])
def parseRef[A](id: Raw[A])(implicit iriDecoder: IriDecoder[A]) = parseIri(id).map(ResourceRef(_))
```

A could be either a String or an Iri as getting a ResourceRef for an Iri never fails, we can map the Iris to a ResourceRef not in the routes but in Resources if needed

Reply:
We need the 3 cases, since `ResourceRef` and `Iri` might not be the same. Imagine someone from some plugin is using this `Resources` and wants to pass a ResourceRef.Tag, not an `Iri`

Reply:
And yes I wanted to call it `IdSegment` precisely because akka http has `Segment`.

Reply:
My thinking about the naming was that so far, it is called by the routes but it could be called by other modules too and Segment might not be so relevant but it is ok (and naming is a hard thing)